pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -190,6 +205,16 @@ type ReconcileHyperConverged struct {
 	ownVersion     string
 	eventEmitter   hcoutil.EventEmitter
 	firstLoop      bool
+
+	// Old v2v CRDs that need to be removed from the cluster.
+	// Removal of the CRDs also leads to removal of its CRs.
+	crdsToRemove []schema.GroupKind
+
+	// Old v2v CRs that need to be removed from the list of related objects.
+	// This list is maintained separately from the list of CRDs to remove,
+	// so we can retry to delete them upon HCO status update failures, even
+	// when the CRD itself has been successfully removed.
+	relatedObjectsToRemove []schema.GroupKind
```

Comment:
I think this is not the best place to put it. Maybe we can have a `crdRemover` type that will hold these lists and all the related code? I think that it should't be part of the recociler.

Reply:
done